dhcpv4/dhcpv4.go

```diff
@@ -144,6 +144,7 @@ func New(modifiers ...Modifier) (*DHCPv4, error) {
 	d := DHCPv4{
 		OpCode:        OpcodeBootRequest,
 		HWType:        iana.HWTypeEthernet,
+		ClientHWAddr:  make(net.HardwareAddr, 6),
```

Comment:
not all hwaddrs are going to have a length of 6. what is this solving?

Reply:
This is the whole commit, which explains the issue in a bit more detail; basically I think that this is a better tradeoff than special-casing it in ToBytes: https://github.com/insomniacslk/dhcp/pull/382/commits/6ce6e224fca409ec3ad6ae8e66795ae3ef2a3e98

New() initializes HWType to Ethernet, for which the length is 6, so at this line, the hwaddr does have a length of 6.
The core issue here is that there are 2 fields that only make sense together, but they can be defined separately (and by default, one of them is undefined), so we resort to hacky special case to detect the default value of the packet and workaround the fact that the default value is broken. 

The best fix would be a generic solution for figuring out the length of the ClientHWAddr based on HWType, but I couldn't find one. I'm not even sure all the public ones are documented or fixed length, and it'd mean we still have no solution for undeclared/non-public HWTypes

So the next idea I had was to at least initialize to a reasonable value and let the user handle the rest if they want to change the HWType and the length.
If you have a better solution to handle this, I also don't love this current solution so I'd be happy to receive some ideas there.